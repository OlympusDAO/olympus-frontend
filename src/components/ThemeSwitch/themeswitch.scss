.toggle-button {
    height: 39px;
    width: 40px;
    border: none !important;
    .moon {
        filter: brightness(0) saturate(100%) invert(70%) sepia(46%) saturate(300%) hue-rotate(173deg) brightness(85%) contrast(84%);
    }
    .sun {
        filter: invert(1);
    }
    svg {
        width: 20px !important;
        height: 20px !important;
    }
}