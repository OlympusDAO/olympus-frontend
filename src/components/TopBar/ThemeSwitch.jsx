import ToggleButton from "@material-ui/lab/ToggleButton";
import SvgIcon from "@material-ui/core/SvgIcon";
import { ReactComponent as SunIcon } from "../../assets/icons/sun.svg";
import { ReactComponent as MoonIcon } from "../../assets/icons/moon.svg";

function ThemeSwitcher({ theme, toggleTheme }) {
  return (
    <ToggleButton
      className="toggle-button"
      type="button"
      title="Change Theme"
      value="check"
      onClick={e => toggleTheme(e)}
    >
      {theme === "dark" ? (
        <SvgIcon component={MoonIcon} color="primary" />
      ) : (
        <SvgIcon component={SunIcon} color="primary" />
      )}
    </ToggleButton>
  );
}

export default ThemeSwitcher;
