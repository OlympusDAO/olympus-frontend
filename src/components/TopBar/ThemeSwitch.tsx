import { t } from "@lingui/macro";
import ToggleButton from "@mui/material/ToggleButton";
import { Icon } from "@olympusdao/component-library";

interface IThemeSwitcherProps {
  theme: string;
  toggleTheme: (e: any) => void;
}

function ThemeSwitcher({ theme, toggleTheme }: IThemeSwitcherProps) {
  return (
    <ToggleButton className="toggle-button" type="button" title={t`Change Theme`} value="check" onClick={toggleTheme}>
      {theme === "dark" ? <Icon name={"moon"} color={"primary"} /> : <Icon name={"sun"} color={"primary"} />}
    </ToggleButton>
  );
}

export default ThemeSwitcher;
