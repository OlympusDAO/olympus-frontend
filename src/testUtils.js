import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { light as lightTheme } from "./themes/light.js";
import { Web3ContextProvider } from "./hooks/web3Context";
import store from "./store.ts";
import { createMemoryHistory } from "history";
import App from "./App";
import { I18nProvider } from "@lingui/react";
import { i18n } from "@lingui/core";

const AllTheProviders = ({ children }) => {
  return (
    <Web3ContextProvider>
      <Provider store={store}>
        <BrowserRouter basename={"/"}>
          <ThemeProvider theme={lightTheme}>
            <CssBaseline />
            {children}
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    </Web3ContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

const renderRoute = function (route) {
  const history = createMemoryHistory();
  history.push(route);
  render(
    <Web3ContextProvider>
      <Provider store={store}>
        <I18nProvider i18n={i18n}>
          <BrowserRouter basename={"/#"}>
            <App />
          </BrowserRouter>
        </I18nProvider>
      </Provider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
