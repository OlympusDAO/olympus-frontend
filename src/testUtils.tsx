import { i18n } from "@lingui/core";
import { I18nProvider } from "@lingui/react";
import CssBaseline from "@material-ui/core/CssBaseline";
import { ThemeProvider } from "@material-ui/core/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React, { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import App from "./App";
import { Web3ContextProvider } from "./hooks/web3Context";
import { ReactQueryProvider } from "./lib/react-query";
import defaultStore from "./store";
import { light as lightTheme } from "./themes/light.js";

const customRender = (ui: ReactElement, store = defaultStore, options?: RenderOptions): RenderResult => {
  const ProviderWrapper = ({ children }: { children?: ReactNode }) => (
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <ThemeProvider theme={lightTheme}>
                <CssBaseline />
                {children}
              </ThemeProvider>
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>
  );
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const renderRoute = function (route: string, store = defaultStore) {
  const history = createMemoryHistory();
  history.push(route);
  return render(
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <App />
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
