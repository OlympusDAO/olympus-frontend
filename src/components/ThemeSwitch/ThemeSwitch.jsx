import React from "react";
import ToggleButton from "@material-ui/lab/ToggleButton";
// import useTheme from "../../hooks/useTheme.js";
import { ReactComponent as SunIcon } from "../../assets/icons/sun-icon.svg";
import { ReactComponent as MoonIcon } from "../../assets/icons/moon-icon.svg";
import "./themeswitch.scss";

function ThemeSwitcher({ theme, toggleTheme }) {
  return (
    <ToggleButton
      className="toggle-button btn top-bar-button btn-overwrite-primer m-2"
      type="button"
      title="Change Theme"
      value="check"
      onClick={e => toggleTheme(e)}
    >
      {theme === "dark" ? <MoonIcon className="moon" /> : <SunIcon className="sun" />}
    </ToggleButton>
  );
}

export default ThemeSwitcher;
