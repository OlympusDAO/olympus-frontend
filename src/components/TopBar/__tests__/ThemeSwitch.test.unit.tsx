import { render } from "../../../testUtils";
import ThemeSwitch from "../ThemeSwitch";

describe("<ThemeSwitch/>", () => {
  it("should render component", () => {
    const { container } = render(<ThemeSwitch theme={"light"} toggleTheme={() => console.log("toggleTheme")} />);
    expect(container).toMatchSnapshot();
  });
});
