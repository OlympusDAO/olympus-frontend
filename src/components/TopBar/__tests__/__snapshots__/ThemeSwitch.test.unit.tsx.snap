// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeSwitch/> should render component 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiToggleButton-root toggle-button"
    tabindex="0"
    title="Change Theme"
    type="button"
    value="check"
  >
    <span
      class="MuiToggleButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 20 20"
      >
        <path
          d="M10.68 2.531a.692.692 0 0 0-.688-.68.69.69 0 0 0-.68.68v1.64c0 .368.313.68.68.68a.692.692 0 0 0 .688-.68v-1.64Zm3.172 2.977a.694.694 0 0 0 0 .96.681.681 0 0 0 .968 0l1.164-1.163a.689.689 0 0 0 0-.97.684.684 0 0 0-.953 0l-1.18 1.173Zm-8.688.96a.679.679 0 0 0 .961 0 .686.686 0 0 0 .008-.96L4.969 4.336a.701.701 0 0 0-.961 0 .701.701 0 0 0-.008.96L5.164 6.47Zm4.828-.124c-2.18 0-3.984 1.804-3.984 3.984 0 2.18 1.804 3.992 3.984 3.992 2.172 0 3.977-1.812 3.977-3.992 0-2.18-1.805-3.984-3.977-3.984Zm7.774 4.664a.692.692 0 0 0 .687-.68.692.692 0 0 0-.687-.68h-1.633a.69.69 0 0 0-.68.68c0 .367.313.68.68.68h1.633ZM2.219 9.648a.69.69 0 0 0-.68.68c0 .367.313.68.68.68h1.633a.692.692 0 0 0 .687-.68.692.692 0 0 0-.687-.68H2.219Zm12.594 4.547a.694.694 0 0 0-.961 0 .694.694 0 0 0 0 .961l1.18 1.172a.679.679 0 0 0 .953-.969l-1.172-1.164ZM4 15.352a.701.701 0 0 0-.008.96.703.703 0 0 0 .969.008l1.164-1.164a.686.686 0 0 0 .008-.96.703.703 0 0 0-.969 0L4 15.351Zm6.68 1.132a.692.692 0 0 0-.688-.68.69.69 0 0 0-.68.68v1.641c0 .367.313.68.68.68a.692.692 0 0 0 .688-.68v-1.64Z"
        />
      </svg>
    </span>
  </button>
</div>
`;
